plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -594,52 +574,88 @@ private DeltaLakeTransactionLogEntry buildTxnEntry(ConnectorSession session, Blo
+                getStringField(txnEntryRow, 0),
+                getLongField(txnEntryRow, 1),
+                getLongField(txnEntryRow, 2));
         log.debug("Result: %s", result);
         return DeltaLakeTransactionLogEntry.transactionEntry(result);
     }
 
     @Nullable
-    private String getString(Block block, int position)
+    private static SqlRow getRowField(SqlRow row, int field)
     {
-        if (block.isNull(position)) {
+        RowBlock valueBlock = (RowBlock) row.getUnderlyingFieldBlock(field);
+        int index = row.getUnderlyingFieldPosition(field);
+        if (valueBlock.isNull(index)) {
             return null;
         }
-        return block.getSlice(position, 0, block.getSliceLength(position)).toString(UTF_8);
+        return valueBlock.getRow(index);
     }
 
-    private long getLong(Block block, int position)
+    @Nullable
+    private static String getStringField(SqlRow row, int field)
     {
-        checkArgument(!block.isNull(position));
-        return block.getLong(position, 0);
+        VariableWidthBlock valueBlock = (VariableWidthBlock) row.getUnderlyingFieldBlock(field);
+        int index = row.getUnderlyingFieldPosition(field);
+        if (valueBlock.isNull(index)) {
+            return null;
+        }
+        return valueBlock.getSlice(index).toStringUtf8();
+    }
+
+    private static long getLongField(SqlRow row, int field)
+    {
+        LongArrayBlock valueBlock = (LongArrayBlock) row.getUnderlyingFieldBlock(field);
+        return valueBlock.getLong(row.getUnderlyingFieldPosition(field));
+    }
+
+    private static int getIntField(SqlRow row, int field)
+    {
+        IntArrayBlock valueBlock = (IntArrayBlock) row.getUnderlyingFieldBlock(field);
+        return valueBlock.getInt(row.getUnderlyingFieldPosition(field));
+    }
+
+    private static OptionalInt getOptionalIntField(SqlRow row, int field)
+    {
+        IntArrayBlock valueBlock = (IntArrayBlock) row.getUnderlyingFieldBlock(field);
+        int index = row.getUnderlyingFieldPosition(field);
+        if (valueBlock.isNull(index)) {
+            return OptionalInt.empty();
+        }
+        return OptionalInt.of(valueBlock.getInt(index));
     }
 
-    private int getInt(Block block, int position)
+    private static boolean getBooleanField(SqlRow row, int field)
     {
-        checkArgument(!block.isNull(position));
-        return block.getInt(position, 0);
+        ByteArrayBlock valueBlock = (ByteArrayBlock) row.getUnderlyingFieldBlock(field);
+        return valueBlock.getByte(row.getUnderlyingFieldPosition(field)) != 0;
     }
 
-    private byte getByte(Block block, int position)
+    private Map<String, String> getMapField(SqlRow row, int field)
     {
-        checkArgument(!block.isNull(position));
-        return block.getByte(position, 0);
+        MapBlock valueBlock = (MapBlock) row.getUnderlyingFieldBlock(field);
+        //noinspection unchecked
```

Comment:
Use `@SuppressWarnings` on the method so that this works for `javac`

Same for the others